main/bamboo/bamboo.go

```diff
@@ -188,63 +188,84 @@ func listenToZookeeper(conf configuration.Configuration, eventBus *event_bus.Eve
 }
 
 func listenToMarathonEventStream(conf *configuration.Configuration, sub api.EventSubscriptionAPI) {
+	ticker := time.NewTicker(1 * time.Second)
+	go listenToMarathonEventStreamLoop(conf, &sub, ticker.C)
+}
+
+type eventBusSink interface {
+	Notify(payload []byte)
+}
+
+func listenToMarathonEventStreamLoop(conf *configuration.Configuration, sink eventBusSink, ticker <-chan time.Time) {
+	for _ = range ticker {
+		for _, marathon := range conf.Marathon.Endpoints() {
+			ch := connectToMarathonEventStream(marathon, conf.Marathon.User, conf.Marathon.Password)
+			for payload := range ch {
+				sink.Notify(payload)
+			}
+		}
+	}
+}
+
+func connectToMarathonEventStream(marathon, user, password string) <-chan []byte {
 	client := &http.Client{}
 	client.Timeout = 0 * time.Second
+	payloadChan := make(chan []byte)
+
+	go func() {
+		defer close(payloadChan)
 
-	for _, marathon := range conf.Marathon.Endpoints() {
-		ticker := time.NewTicker(1 * time.Second)
 		eventsURL := marathon + "/v2/events"
-		go func() {
-			for _ = range ticker.C {
-				req, err := http.NewRequest("GET", eventsURL, nil)
-				req.Header.Set("Accept", "text/event-stream")
-				if len(conf.Marathon.User) > 0 && len(conf.Marathon.Password) > 0 {
-					req.SetBasicAuth(conf.Marathon.User, conf.Marathon.Password)
-				}
-				if err != nil {
-					errorMsg := "An error occurred while creating request to Marathon events system: %s\n"
-					log.Printf(errorMsg, err)
-					continue
-				}
+		req, err := http.NewRequest("GET", eventsURL, nil)
+		if err != nil {
+			errorMsg := "An error occurred while creating request to Marathon events system: %s\n"
+			log.Printf(errorMsg, err)
+			return
+		}
 
-				resp, err := client.Do(req)
-				if err != nil {
-					errorMsg := "An error occurred while making a request to Marathon events system: %s\n"
+		req.Header.Set("Accept", "text/event-stream")
+		if len(user) > 0 && len(password) > 0 {
+			req.SetBasicAuth(user, password)
+		}
+
+		resp, err := client.Do(req)
+		if err != nil {
+			errorMsg := "An error occurred while making a request to Marathon events system: %s\n"
+			log.Printf(errorMsg, err)
+			return
+		}
+
+		defer resp.Body.Close()
+
+		reader := bufio.NewReader(resp.Body)
+		for {
+			line, err := reader.ReadString('\n')
+			if err != nil {
+				if err != io.EOF {
+					errorMsg := "An error occurred while reading Marathon event: %s\n"
 					log.Printf(errorMsg, err)
```

Comment:
Dunno why we're not doing `log.Printf("An error occurred while reading Marathon event: %s\n", err)` here. I guess it's better to be consistent for now though